pkg/db/interface.go

```diff
@@ -132,56 +135,74 @@ func NewWithSQLConn(db *sql.DB) (VizierDBInterface, error) {
 	return d, nil
 }
 
-func New() (VizierDBInterface, error) {
-	db, err := openSQLConn(dbDriver, getDbName(), connectInterval, connectTimeout)
+func (d *dbConn) GetStudyMetrics(id string) ([]string, error) {
+	row := d.db.QueryRow("SELECT metrics FROM studies WHERE id = ?", id)
+	var metrics string
+	err := row.Scan(&metrics)
 	if err != nil {
-		return nil, fmt.Errorf("DB open failed: %v", err)
+		return nil, err
 	}
-	return NewWithSQLConn(db)
+	retMetrics := strings.Split(metrics, ",\n")
+	return retMetrics, nil
 }
 
-func generateRandid() string {
-	// UUID isn't quite handy in the Go world
-	id := make([]byte, 8)
-	_, err := rand.Read(id)
+func (d *dbConn) GetStudyIDsTypesList() ([]string, []string, error) {
```

Comment:
I did not find where GetStudyIDsTypesList is used. Maybe you can delete it.
BTW,  if you still persist to remain it. I think you can use "map[string]string" to replace "[]string, []string" in return where key is studyid and value is type

Reply:
Resolved